### PR TITLE
Update dependencies and workaround dotnet/sdk#2976

### PR DESCRIPTION
**Pull request by natemcmaster**
Opened manually using our script that finds coherent dependencies

### CODE REVIEW

**Comment by benaadams** [comment]
```
Components/Blazor/Blazor/test/Hosting/WebAssemblyHostTest.cs(65,63):
 error CS0117: 'JSRuntime' does not contain a definition for 'Current'
 [/home/vsts/work/1/s/src/Components/Blazor/Blazor/test/Microsoft.AspNetCore.Blazor.Tests.csproj]
Components/Blazor/testassets/Microsoft.AspNetCore.Blazor.E2EPerformance/BenchmarkEvent.cs(12,45):
 error CS0117: 'JSRuntime' does not contain a definition for 'Current' [/home/vsts/work/1/s/src/Components/Blazor/testassets/Microsoft.AspNetCore.Blazor.E2EPerformance/Microsoft.AspNetCore.Blazor.E2EPerformance.csproj]
```

**Comment by natemcmaster** [comment]
@pranavkm handing the mic to you. Can you update this repo to react to https://github.com/aspnet/Extensions/pull/1118?

cc @rynowak @SteveSandersonMS

**Comment by benaadams** [comment]
Back to `LoggerFactory` being disposed
```
 System.ObjectDisposedException: Cannot access a disposed object.
  Object name: 'LoggerFactory'.
     at Microsoft.Extensions.Logging.LoggerFactory.CreateLogger(String categoryName)
     at Microsoft.EntityFrameworkCore.Internal.ExternalLoggerFactory.CreateLogger(String categoryName)
     at Microsoft.EntityFrameworkCore.Internal.DiagnosticsLogger`1..ctor(ILoggerFactory loggerFactory, ILoggingOptions loggingOptions, DiagnosticSource diagnosticSource)
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
     at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
     at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
     at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
     at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
     at Microsoft.EntityFrameworkCore.DbContext.Set[TEntity]()
     at Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`9.get_UsersSet() in /_/src/Identity/EntityFrameworkCore/src/UserStore.cs:line 128
     at Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`9.get_Users() in /_/src/Identity/EntityFrameworkCore/src/UserStore.cs:line 262
     at Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`9.FindByNameAsync(String normalizedUserName, CancellationToken cancellationToken) in /_/src/Identity/EntityFrameworkCore/src/UserStore.cs:line 254
     at Microsoft.AspNetCore.Identity.UserManager`1.FindByNameAsync(String userName) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 573
     at Microsoft.AspNetCore.Identity.UserValidator`1.ValidateUserName(UserManager`1 manager, TUser user, ICollection`1 errors) in /_/src/Identity/Extensions.Core/src/UserValidator.cs:line 79
     at Microsoft.AspNetCore.Identity.UserValidator`1.ValidateAsync(UserManager`1 manager, TUser user) in /_/src/Identity/Extensions.Core/src/UserValidator.cs:line 56
     at Microsoft.AspNetCore.Identity.UserManager`1.ValidateUserAsync(TUser user) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 2494
     at Microsoft.AspNetCore.Identity.UserManager`1.CreateAsync(TUser user) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 470
     at Microsoft.AspNetCore.Identity.UserManager`1.CreateAsync(TUser user, String password) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 602
     at Microsoft.AspNetCore.Identity.UI.V3.Pages.Account.Internal.RegisterModel`1.OnPostAsync(String returnUrl) in /_/src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Register.cshtml.cs:line 125
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Convert[T](Object taskAsObject) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/ExecutorFactory.cs:line 145
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Execute(Object receiver, Object[] arguments) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/ExecutorFactory.cs:line 139
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeHandlerMethodAsync() in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 288
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeNextPageFilterAsync() in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 654
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Rethrow(PageHandlerExecutedContext context) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 696
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 628
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeInnerFilterAsync() in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 93
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeNextResourceFilter() in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 809
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContext context) in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 1153
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 762
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync() in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 129
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeAsync() in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 101
     at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointMiddleware.cs:line 53
     at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 79
     at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 62
```

**Comment by benaadams** [comment]
e.g.
```diff
-  { typeof(ILoggerFactory), new ServiceCharacteristics(ServiceLifetime.Singleton) },
+  { typeof(ILoggerFactory), new ServiceCharacteristics(ServiceLifetime.Scoped) },
```

**Comment by benaadams** [comment]
Back to `LoggerFactory` being disposed
```
 System.ObjectDisposedException: Cannot access a disposed object.
  Object name: 'LoggerFactory'.
     at Microsoft.Extensions.Logging.LoggerFactory.CreateLogger(String categoryName)
     at Microsoft.EntityFrameworkCore.Internal.ExternalLoggerFactory.CreateLogger(String categoryName)
     at Microsoft.EntityFrameworkCore.Internal.DiagnosticsLogger`1..ctor(ILoggerFactory loggerFactory, ILoggingOptions loggingOptions, DiagnosticSource diagnosticSource)
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
     at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
     at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
     at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
     at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
     at Microsoft.EntityFrameworkCore.DbContext.Set[TEntity]()
     at Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`9.get_UsersSet() in /_/src/Identity/EntityFrameworkCore/src/UserStore.cs:line 128
     at Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`9.get_Users() in /_/src/Identity/EntityFrameworkCore/src/UserStore.cs:line 262
     at Microsoft.AspNetCore.Identity.EntityFrameworkCore.UserStore`9.FindByNameAsync(String normalizedUserName, CancellationToken cancellationToken) in /_/src/Identity/EntityFrameworkCore/src/UserStore.cs:line 254
     at Microsoft.AspNetCore.Identity.UserManager`1.FindByNameAsync(String userName) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 573
     at Microsoft.AspNetCore.Identity.UserValidator`1.ValidateUserName(UserManager`1 manager, TUser user, ICollection`1 errors) in /_/src/Identity/Extensions.Core/src/UserValidator.cs:line 79
     at Microsoft.AspNetCore.Identity.UserValidator`1.ValidateAsync(UserManager`1 manager, TUser user) in /_/src/Identity/Extensions.Core/src/UserValidator.cs:line 56
     at Microsoft.AspNetCore.Identity.UserManager`1.ValidateUserAsync(TUser user) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 2494
     at Microsoft.AspNetCore.Identity.UserManager`1.CreateAsync(TUser user) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 470
     at Microsoft.AspNetCore.Identity.UserManager`1.CreateAsync(TUser user, String password) in /_/src/Identity/Extensions.Core/src/UserManager.cs:line 602
     at Microsoft.AspNetCore.Identity.UI.V3.Pages.Account.Internal.RegisterModel`1.OnPostAsync(String returnUrl) in /_/src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Register.cshtml.cs:line 125
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Convert[T](Object taskAsObject) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/ExecutorFactory.cs:line 145
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Execute(Object receiver, Object[] arguments) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/ExecutorFactory.cs:line 139
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeHandlerMethodAsync() in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 288
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeNextPageFilterAsync() in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 654
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Rethrow(PageHandlerExecutedContext context) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 696
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 628
     at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeInnerFilterAsync() in /_/src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionInvoker.cs:line 93
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeNextResourceFilter() in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 809
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContext context) in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 1153
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 762
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync() in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 129
     at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeAsync() in /_/src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs:line 101
     at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointMiddleware.cs:line 53
     at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext) in /_/src/Http/Routing/src/EndpointRoutingMiddleware.cs:line 79
     at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in /_/src/Security/Authentication/Core/src/AuthenticationMiddleware.cs:line 62
```

Could "Make ILoggerFactory a scoped service" https://github.com/aspnet/EntityFrameworkCore/pull/14706 have caused the disposal? /cc @ajcvickers

e.g.
```diff
-  { typeof(ILoggerFactory), new ServiceCharacteristics(ServiceLifetime.Singleton) },
+  { typeof(ILoggerFactory), new ServiceCharacteristics(ServiceLifetime.Scoped) },
```
Or perhaps
```diff
- services.AddSingleton(loggerFactory);
+ services.AddScoped<ILoggerFactory>(_ => new ExternalLoggerFactory(loggerFactory));
```

**Comment by benaadams** [comment]
Perhaps is causing both `ILoggerFactory`'s to get disposed? https://github.com/aspnet/Extensions/issues/1155

**Comment by benaadams** [comment]
Most of the 368 failed tests tests are under `Microsoft.AspNetCore.Identity.FunctionalTests`

Couple in `Microsoft.Extensions.Diagnostics.HealthChecks.DbContextHealthCheckTest` which seem different

**Comment by ajcvickers** [comment]
In have a PR out to fix EF: https://github.com/aspnet/EntityFrameworkCore/pull/14783

**Review by dougbu — COMMENTED** [review]
Once we have the newer EF Core and the builds turn green :shipit:

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -9,380 +9,380 @@
...
+      <Sha>b07bc5efcad439d5157f65df176e08f26ccb4b88</Sha>
     </Dependency>
-    <Dependency Name="Internal.AspNetCore.Analyzers" Version="3.0.0-preview3.19115.4">
+    <Dependency Name="Internal.AspNetCore.Analyzers" Version="3.0.0-preview3.19120.1">
       <Uri>https://github.com/aspnet/Extensions</Uri>
-      <Sha>321f30556d63ec8af6eb0bc43d534b6b2e5458a0</Sha>
+      <Sha>97d7856ea482015209d60cf8955bacf8053f4cb5</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.Testing" Version="3.0.0-preview3.19115.4">
+    <Dependency Name="Microsoft.AspNetCore.Testing" Version="3.0.0-preview3.19120.1">
```
[the remark this answers is not in the document]
Comment (natemcmaster) [reply eng/Version.Details.xml]: This repo is already using xunit 2.4.0. I think aspnetcore-tooling was the only one still on 2.3

**Comment by natemcmaster** [comment]
Pushed a change to react to https://github.com/aspnet/AspNetCore-Tooling/pull/240, https://github.com/aspnet/EntityFrameworkCore/pull/14790, and  https://github.com/aspnet/Extensions/pull/1159

File: build/sources.props [hunk 2]
```diff
@@ -10,6 +10,9 @@
 
     <RestoreSources Condition=" '$(DotNetBuildOffline)' != 'true' ">
       $(RestoreSources);
+      https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json;
+      https://dotnetfeed.blob.core.windows.net/aspnet-entityframeworkcore/index.json;
+      https://dotnetfeed.blob.core.windows.net/aspnet-aspnetcore-tooling/index.json;
```
Comment (dougbu) [inline build/sources.props]: Doesn't this repo also publish to https://dotnetfeed.blob.core.windows.net? I'm not seeing a change there.

**Review by dougbu — APPROVED** [review]

File: build/sources.props (hunk 2 above)
Comment (natemcmaster) [reply build/sources.props]: Can you clarify? Not sure I understand....

image

**Comment by benaadams** [comment]
Not sure it has the new EF stuff yet?
```
       An unhandled exception has occurred while executing the request.
  System.ObjectDisposedException: Cannot access a disposed object.
  Object name: 'LoggerFactory'.
     at Microsoft.Extensions.Logging.LoggerFactory.CreateLogger(String categoryName)
     at Microsoft.EntityFrameworkCore.Internal.ExternalLoggerFactory.CreateLogger(String categoryName)
     at Microsoft.EntityFrameworkCore.Internal.DiagnosticsLogger`1..ctor(ILoggerFactory loggerFactory, ILoggingOptions loggingOptions, DiagnosticSource diagnosticSource)
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
     at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
     at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
     at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
     at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
     at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
     at Microsoft.EntityFrameworkCore.DbContext.Set[TEntity]()
```

**Comment by dougbu** [comment]
@benaadams no it doesn't. @natemcmaster could you rerun your tool please?

File: build/sources.props (hunk 2 above)
Comment (dougbu) [reply build/sources.props]: The "Change publish feed to workaround 'Waiting to obtain an exclusive lock on the feed.'" PRs in other repos also updated azure-pipelines.yml. I was expecting a similar change in .azure/pipelines/ci.yml Now, I suspect that expectation was off base.

Comment (natemcmaster) [reply build/sources.props]: Oh, I understand. Yes, I'll add the appropriate change for this repo. It's in a different spot, not azure-pipelines.yml. Good catch.

**Comment by natemcmaster** [comment]
Updated with latest EFCore. 🤞

**Comment by benaadams** [comment]
MVC using its own obsolete method?
```
/home/vsts/work/1/s/src/Mvc/Mvc.Localization/src/HtmlLocalizer.cs(95,38):

 error CS0618: 'IStringLocalizer.WithCulture(CultureInfo)' is obsolete:
 'This method is obsolete. Use `CurrentCulture` and `CurrentUICulture` instead.' 
[/home/vsts/work/1/s/src/Mvc/Mvc.Localization/src/Microsoft.AspNetCore.Mvc.Localization.csproj]
```

**Comment by natemcmaster** [comment]
@ryanbrandenburg can you update AspNetCore to react to change you made in Extensions?

**Review by dougbu — APPROVED** [review]

**Comment by benaadams** [comment]
Restart Build Linux x64; was a docker registry timeout
```
Step 1/12 : FROM microsoft/dotnet:2.1-runtime-deps-bionic
Get https://registry-1.docker.io/v2/: net/http: TLS handshake timeout
##[error]Bash exited with code '1'.
```

**Comment by benaadams** [comment]
macOs failure
```
Microsoft.Extensions.Diagnostics.HealthChecks.DbContextHealthCheckTest.CheckAsync_CustomTest_Unhealthy
Assert.Equal() Failure
Expected: Unhealthy
Actual: Healthy

Stack trace
   at Microsoft.Extensions.Diagnostics.HealthChecks.DbContextHealthCheckTest.CheckAsync_CustomTest_Unhealthy() in /_/src/Middleware/HealthChecks.EntityFrameworkCore/test/DbContextHealthCheckTest.cs:line 103
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by natemcmaster** [comment]
Getting closer. The ILoggerFactory test failures and compiler errors are gone.

Next failure:
```
  Failed   Microsoft.Extensions.Diagnostics.HealthChecks.DbContextHealthCheckTest.CheckAsync_CustomTest_Unhealthy
  Error Message:
   Assert.Equal() Failure
  Expected: Unhealthy
  Actual:   Healthy
```

@Tratcher @rynowak any ideas?

**Comment by ajcvickers** [comment]
@natemcmaster @Tratcher @rynowak I don't have time to investigate right now, but ping me if the `DbContextHealthCheck` looks like it is caused by EF changes.

**Comment by natemcmaster** [comment]
There's a more than good chance the failure is due to EFCore. The health checks stuff hasn't changed recently. I'll take a look at the test to see if the test makes bad assumptions about EFCore and service scopes.

**Comment by natemcmaster** [comment]
@ajcvickers it looks like the in-memory database used in this test returns different results.

Before EFCore update:

image

After EFCore update:

image


Test code: https://github.com/aspnet/AspNetCore/blob/90e893ed077b82168354b6563f40bebad398ed77/src/Middleware/HealthChecks.EntityFrameworkCore/test/DbContextHealthCheckTest.cs#L85-L117

**Comment by natemcmaster** [comment]
@ajcvickers @rynowak  - workaround for changes in EF in-memory database behavior:

https://github.com/aspnet/AspNetCore/pull/7844/commits/aaaecbf66aeb9f2dfbe77a8f5f052b95834222d1

**Comment by rynowak** [comment]
Looks fine, a comment would make it mo better.

**Comment by ajcvickers** [comment]
@natemcmaster @rynowak Thanks Nate. Here's what's happening. When an in-memory database with a specific name is used, that database is under normal circumstances supposed to be the same database later if the same name is used. However, in reality the named database is rooted by default in EF's internal service provider. So the above behavior holds true as long as the same internal service provider is used by EF, which for a given EF configuration should be true.

So this behavior that is now being seen in these tests--when a database called "Test" is used it is shared by all tests that create a database with that name. This is as things are supposed to be, and is one of the things that the EF breaking change makes "better". (The reason for this is that previously anytime a new ILoggerFactory was used with each DbContext instance, it caused EF to build a new internal service provider for that DbContext instance. (This is easy to hit in ASP.NET tests that often create a new ILoggerFactory for each test.) This no longer happens, so all the tests use the same EF internal service provider, which means they get to share the in-memory database.)

In this case, the tests were dependent on the old behavior. So this is a case where the breaking change really broke things, but in an expected way, and with ultimately better, more consistent behavior. Still we should make sure it's documented. /cc @divega

**Pull request merged**